Phylum/script/load_threat_data.py

```diff
@@ -0,0 +1,239 @@
+        self.phylum_token = phylum_apitoken
+        # self.sl_threats = []
+
+    def get_threatfeed_folder(self):
+        personal = self.sumo.get_personal_folder().json()
+        children = personal.get('children')
+        phylum_folder_id = ""
+        for elem in children:
+            if elem.get('name') == "phylum-threat-data":
+                phylum_folder_id = elem.get('id')
+                print(f"[*] Found phylum-threat-data folder: {phylum_folder_id}")
+
+        if phylum_folder_id != "":
+            self.phylum_folder_id = phylum_folder_id
+        else:
+            sys.exit(1)
+
+    # Did this manually
+    # def create_lookup_table(self, name):
+
+    def get_phylum_lookup_table(self, path):
+        lookup_table = self.sumo.get_content_item_by_path(path).json()
+        self.lookup_table_id = lookup_table.get('id')
+        print(f"[*] Found lookup table ID: {self.lookup_table_id}")
+        return
+
+    # def get_phylum_token(self):
+    #     get_token_command = ["phylum", "auth", "token", "-b"]
+    #     process = run(get_token_command, text=True, capture_output=True)
+    #     if process.returncode != 0:
+    #         print(f"Get Token Command Failed: {process.returncode}")
+    #         sys.exit(1)
+    #     else:
+    #         print(f"[*] Got Phylum Access Token")
+    #         self.phylum_token = process.stdout.strip()
+    #     return
+
+    def get_phylum_threat_data(self):
+        page = 1
+        url = f"https://threats.phylum.io?per_page=50&page={page}"
+        token_str = f"Bearer {self.phylum_token}"
+        headers = {'Authorization': token_str}
+        resp = requests.get(url, headers=headers)
+        count = 1
+
+        packages = []
+
+        if resp.status_code != 200:
+            print(f"Failed to call threat feed: {resp.status_code}")
+            sys.exit(1)
+
+        data = resp.json() 
+        packages.extend(data.get('packages'))
+
+        while data.get('has_next') == True:
+            page += 1
+            url = f"https://threats.phylum.io?per_page=50&page={page}"
+            token_str = f"Bearer {self.phylum_token}"
+            headers = {'Authorization': token_str}
+            resp = requests.get(url, headers=headers)
+            data = resp.json() 
+            packages.extend(data.get('packages'))
+
+            #has_more = data.get('has_next')
+            # while has_more:
+            #     count += 1
+            #     add_url = f"{url}&page={count}"
+            #     add_resp = requests.get(add_url, headers=headers)
+            #     if add_resp.status_code != 200:
+            #         print(f"Status Code = {add_resp.status_code}")
+            #         break
+            #     add_data = add_resp.json()
+            #     data['packages'].extend(add_data['packages'])
+            #     has_more = add_data.get('has_next')
+            #     # if count > 100:
+            #     #     break
+        return packages
```

Comment:
the maximum size of lookup table is 100MB can we put a warning or assert statement